File: app/src/main/java/com/arny/myapidemo/mvp/auth/AuthView.java
package com.arny.myapidemo.mvp.auth;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;
@StateStrategyType(AddToEndSingleStrategy.class)
public interface AuthView extends MvpView {
	void loggedIn();

	void showLoginError();

	void showPassError();

    void hideErrors();

	void showAuthError();

	void showProgress();

	void hideProgress();

	void setToolbarTitle(String title);

	void updateProgress(String progress);

	void showAlert(String message);

	void hideMessage();
}

File: app/src/main/java/com/arny/myapidemo/mvp/auth/AuthView.kt
package com.arny.myapidemo.mvp.auth

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

@StateStrategyType(AddToEndSingleStrategy::class)
interface AuthView : MvpView {
    fun signUp()

    fun signIn()

    fun showLoginError()

    fun showPassError()

    fun hideErrors()

    fun showAuthError()

    fun showProgress()

    fun hideProgress()

    fun setToolbarTitle(title: String)

    fun updateProgress(progress: String)

    fun showAlert(message: String)

    fun hideMessage()
}
